tests/paragraph.js

```diff
@@ -52,14 +65,265 @@ describe( 'Paragraph feature', () => {
 			expect( getModelData( doc, { withoutSelection: true } ) ).to.equal( '<paragraph>foo</paragraph><paragraph>baz</paragraph>' );
 			expect( editor.getData() ).to.equal( '<p>foo</p><p>baz</p>' );
 		} );
+
+		describe( 'generic text converter (text autoparagraphing)', () => {
+			it( 'should autoparagraph text', () => {
+				editor.setData( 'foo' );
+
+				expect( getModelData( doc, { withoutSelection: true } ) ).to.equal( '<paragraph>foo</paragraph>' );
+				expect( editor.getData() ).to.equal( '<p>foo</p>' );
+			} );
+
+			it( 'should not autoparagraph text (in clipboard holder)', () => {
+				const modelFragment = editor.data.parse( 'foo', '$clipboardHolder' );
+
+				expect( stringifyModel( modelFragment ) )
+					.to.equal( 'foo' );
+			} );
+
+			it( 'should not autoparagraph text (in a context which does not allow paragraphs', () => {
+				doc.schema.registerItem( 'specialRoot' );
+
+				const modelFragment = editor.data.parse( 'foo', 'specialRoot' );
+
+				expect( stringifyModel( modelFragment ) )
+					.to.equal( '' );
+			} );
+
+			it( 'should autoparagraph text next to allowed element', () => {
+				doc.schema.registerItem( 'heading1', '$block' );
+				buildViewConverter().for( editor.data.viewToModel ).fromElement( 'h1' ).toElement( 'heading1' );
+
+				const modelFragment = editor.data.parse( '<h1>foo</h1>bar<p>bom</p>' );
+
+				expect( stringifyModel( modelFragment ) )
+					.to.equal( '<heading1>foo</heading1><paragraph>bar</paragraph><paragraph>bom</paragraph>' );
+			} );
+
+			it( 'should autoparagraph 3 inline inline nodes into one paragraph', () => {
```

Comment:
As you will be making changes to tests already, remove double "inline" word too, please.